src/helpers/pull-helpers/get-task-spec.ts

```diff
@@ -0,0 +1,43 @@
+import { Context } from "../../types";
+import { getOwnerRepoIssueNumberFromUrl } from "../get-owner-repo-issue-from-url";
+import { checkIfPrClosesIssues } from "../gql-functions";
+
+export async function getTaskNumberFromPullRequest(context: Context<"pull_request.opened" | "pull_request.ready_for_review">) {
+  const {
+    payload: { pull_request },
+    logger,
+  } = context;
+  let issueNumber;
+
+  const { issues: closingIssues } = await checkIfPrClosesIssues(context.octokit, {
+    owner: pull_request.base.repo.owner.login,
+    repo: pull_request.base.repo.name,
+    pr_number: pull_request.number,
+  });
+
+  if (closingIssues.length === 0) {
+    const linkedViaBodyHash = pull_request.body?.match(/#(\d+)/g);
+    const urlMatch = getOwnerRepoIssueNumberFromUrl(pull_request.body);
+
+    if (linkedViaBodyHash?.length) {
+      issueNumber = Number(linkedViaBodyHash[0].replace("#", ""));
+    }
+
+    if (urlMatch && !issueNumber) {
+      issueNumber = Number(urlMatch.issueNumber);
+    }
```

Comment:
If there's no linked issues than the author didn't use `Resolves #123` phrase which is wrong so we should convert PR to draft and post a comment something like `You need to link an issue and after that convert the PR to ready for review`. @0x4007 rfc

Reply:
I thought we have more robust SDK methods to detect if it is linked without having to rely only on the pull body text. 

I have mixed feelings because we also need to accommodate non coding tasks which doesn't make use of pulls but as long as that's handled then it's fine.

Reply:
i think we can just post a comment because on a non coding task, the pull reviewing bot is not gonna get called anyways because there wouldnt be a pr attached